storage.go

```diff
@@ -112,17 +112,18 @@ func (s *Storage) read(ctx context.Context, path string, w io.Writer, opt pairSt
 		}
 	}
 
-	var rc io.Reader
+	if opt.HasSize {
```

Comment:
It looks buggy if we have size and IoCallback at the same time.

Reply:
Yes, but I think on the one hand, the test may also need some improvement. For example, can we achieve coverage from multiple angles at the same time, otherwise it is possible to add a combination of different possibilities again

Reply:
Nice idea, we will implement this in go-integration-tests.